eos_api.hpp

```diff
@@ -1341,377 +1332,343 @@ uint32_t oraclize_getSenderNonce(account_name sender)
     return nonce;
 }
 
-checksum256 oraclize_getCommitment(uint64_t shortqueryid)
-{
-    ds_scommitment last_commitments(N(_self), N(_self));
-    auto itr = last_commitments.find(shortqueryid);
-    checksum256 commitment;
-    if (itr != last_commitments.end())
-    {
-        commitment = itr->commitment;
-    }
-    return commitment;
-}
-
-checksum256 oraclize_getQueryId(uint64_t shortqueryid)
-{
-    ds_scommitment last_commitments(N(_self), N(_self));
-    auto itr = last_commitments.find(shortqueryid);
-    checksum256 queryid;
-    if (itr != last_commitments.end())
-    {
-        queryid = itr->queryid;
-    }
-    return queryid;
-}
-
-checksum256 oraclize_getNextQueryId(account_name sender)
+capi_checksum256 __oraclize_getNextQueryId(const name sender)
 {
     // Get values to generate the queryId
-    uint32_t nonce = oraclize_getSenderNonce(sender);
-    size_t tx_size = transaction_size();
+    const uint32_t nonce = __oraclize_getSenderNonce(sender);
+    const size_t tx_size = transaction_size();
     // calculate the hash of the previous values
-    uint8_t tbh[sizeof(sender) + sizeof(nonce) + sizeof(tx_size)]; //account_name[uint64_t] + nonce[uint32_t]
+    uint8_t tbh[sizeof(sender) + sizeof(nonce) + sizeof(tx_size)];
     std::memcpy(tbh, &sender, sizeof(sender));
     std::memcpy(tbh + sizeof(sender), &nonce, sizeof(nonce));
     std::memcpy(tbh + sizeof(sender) + sizeof(nonce), &tx_size, sizeof(tx_size));
-    checksum256 calc_hash;
+    capi_checksum256 calc_hash;
     sha256((char *)tbh, sizeof(tbh), &calc_hash);
     return calc_hash;
 }
 
-checksum256 oraclize_getQueryId_local__(checksum256 queryId, account_name sender)
+capi_checksum256 __oraclize_getQueryId_local(const capi_checksum256 queryId, const name sender)
 {
-  // retreive the short queryId from the passed 
-  uint64_t myQueryId_short;
-  std::memcpy(&myQueryId_short, &queryId.hash, sizeof(myQueryId_short));
+    // retreive the short queryId from the passed 
+    uint64_t myQueryId_short;
+    std::memcpy(&myQueryId_short, &queryId.hash, sizeof(myQueryId_short));
     
-  // access the local table 
-  ds_queryid queryids(sender, sender); 
-  auto itr = queryids.find(myQueryId_short);
-  checksum256 queryId_expected;
-  std::memcpy(&queryId_expected, 0, sizeof(queryId_expected));
-  if (itr != queryids.end())
-  {
-    queryId_expected = itr->qid;  
-  }
-  return queryId_expected;
+    // access the local table 
+    ds_queryid queryids(sender, sender.value); 
+    const auto itr = queryids.find(myQueryId_short);
+    capi_checksum256 queryId_expected;
+    std::memcpy(&queryId_expected, 0, sizeof(queryId_expected));
+    if (itr != queryids.end())
+    {
+        queryId_expected = itr->qid;
+    }
+    return queryId_expected;
 }
 
-void oraclize_emplaceQueryId_local__(checksum256 myQueryId, account_name sender)
+void __oraclize_emplaceQueryId_local(const capi_checksum256 myQueryId, const name sender)
```

Comment:
rename to "__oraclize_queryId_emplace"

Reply:
Changed in `oraclize_queryId_localEmplace`, to specify that the `emplace` operation will cost users some RAM, since it is a local table storage.